src/main/java/de/tum/in/www1/artemis/web/rest/metis/conversation/ConversationResource.java

```diff
@@ -0,0 +1,167 @@
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import de.tum.in.www1.artemis.domain.metis.conversation.Channel;
+import de.tum.in.www1.artemis.domain.metis.conversation.Conversation;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.dto.UserPublicInfoDTO;
+import de.tum.in.www1.artemis.service.metis.conversation.ConversationService;
+import de.tum.in.www1.artemis.service.metis.conversation.ConversationService.ConversationMemberSearchFilters;
+import de.tum.in.www1.artemis.service.metis.conversation.auth.ChannelAuthorizationService;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.metis.conversation.dtos.ConversationDTO;
+import de.tum.in.www1.artemis.web.rest.metis.conversation.dtos.ConversationUserDTO;
+import tech.jhipster.web.util.PaginationUtil;
+
+@RestController
+@RequestMapping("/api/courses")
+public class ConversationResource {
+
+    private final Logger log = LoggerFactory.getLogger(ConversationResource.class);
+
+    private final ConversationService conversationService;
+
+    private final ChannelAuthorizationService channelAuthorizationService;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    private final UserRepository userRepository;
+
+    private final CourseRepository courseRepository;
+
+    public ConversationResource(ConversationService conversationService, ChannelAuthorizationService channelAuthorizationService,
+            AuthorizationCheckService authorizationCheckService, UserRepository userRepository, CourseRepository courseRepository) {
+        this.conversationService = conversationService;
+        this.channelAuthorizationService = channelAuthorizationService;
+        this.authorizationCheckService = authorizationCheckService;
+        this.userRepository = userRepository;
+        this.courseRepository = courseRepository;
+    }
+
+    /**
+     * GET /api/courses/:courseId/conversations: Returns all conversations of a course where the requesting user is a member
+     *
+     * @param courseId the id of the course
+     * @return ResponseEntity with status 200 (OK) and with body containing the list of conversations where the requesting user is a member
+     */
+    @GetMapping("/{courseId}/conversations")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<List<ConversationDTO>> getConversationsOfUser(@PathVariable Long courseId) {
+        var requestingUser = this.userRepository.getUserWithGroupsAndAuthorities();
```

Comment:
We normally do not add `this` when the method signature does not have a conflicting parameter name (applies to the whole class and everywhere else).
```suggestion
        var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
```